tests/components/matter/test_water_heater.py

```diff
@@ -129,12 +129,73 @@ async def test_water_heater_set_operation_mode(
         )
 
     @pytest.mark.parametrize("node_fixture", ["silabs_water_heater"])
-    async def test_water_heater_set_on_off(
+    async def test_water_heater_boostmode(
         hass: HomeAssistant,
         matter_client: MagicMock,
         matter_node: MatterNode,
     ) -> None:
         """Test water_heater set operation mode service."""
+        # Boost 1h (3600s)
+        boost_info: type[
+            clusters.WaterHeaterManagement.Structs.WaterHeaterBoostInfoStruct
+        ] = clusters.WaterHeaterManagement.Structs.WaterHeaterBoostInfoStruct(
+            duration=3600
+        )
+        state = hass.states.get("water_heater.water_heater")
+        assert state
+
+        # enable water_heater boostmode
+        await hass.services.async_call(
+            "water_heater",
+            "set_operation_mode",
+            {
+                "entity_id": "water_heater.water_heater",
+                "operation_mode": STATE_HIGH_DEMAND,
+            },
+            blocking=True,
+        )
+
+        state = hass.states.get("water_heater.water_heater")
+        assert state.state == STATE_HIGH_DEMAND
```

Comment:
I don't believe it matters here what the state is, since we haven't triggered any state update with a subscription callback. The service action call itself doesn't change the state, right?

Check all the tests for this pattern if we should update it.